docker/puppet-agent-alpine/Dockerfile

```diff
@@ -180,36 +183,43 @@ LABEL org.label-schema.maintainer="Puppet Release Team <release@puppet.com>" \
       org.label-schema.schema-version="1.0" \
       org.label-schema.dockerfile="/Dockerfile"
 
+# boost is in main, while yaml-cpp is in community
+RUN apk add \
+    --no-cache \
+    --update-cache \
+    --repository http://nl.alpinelinux.org/alpine/v3.10/main \
+    --repository http://nl.alpinelinux.org/alpine/v3.10/community \
+    boost-date_time=1.69.0-r4 \
+    boost-doc=1.69.0-r4 \
+    boost-filesystem=1.69.0-r4 \
+    boost-graph=1.69.0-r4 \
+    boost-iostreams=1.69.0-r4 \
+    boost-locale=1.69.0-r4 \
+    boost-log=1.69.0-r4 \
+    boost-math=1.69.0-r4 \
+    boost-prg_exec_monitor=1.69.0-r4 \
+    boost-program_options=1.69.0-r4 \
+    boost-random=1.69.0-r4 \
+    boost-regex=1.69.0-r4 \
+    boost-serialization=1.69.0-r4 \
+    boost-system=1.69.0-r4 \
+    boost-thread=1.69.0-r4 \
+    boost-unit_test_framework=1.69.0-r4 \
```

Comment:
We might still be carrying a few unnecessary boost libraries, but leaving the prior art for now.